app/controllers/application_controller.rb

```diff
@@ -3,10 +3,16 @@ class ApplicationController < ActionController::API
   include ActiveStorage::SetCurrent
 
   rescue_from ActiveRecord::RecordInvalid, with: :render_record_invalid
+  rescue_from ActiveRecord::RecordNotFound, with: :render_record_not_found
 
   private
 
   def render_record_invalid(exception)
     render json: { errors: exception.record.errors.as_json }, status: :unprocessable_entity
   end
+
+  def render_record_not_found(exception)
+    render json: { errors: "Could not find #{exception.model} with id #{exception.id}" },
```

Comment:
I don't think returning a custom message uncluding the id is necessary given that the request should already receive the id as param, so it is known already by the requester